filecoin-proofs/src/bin/parampublish.rs

```diff
@@ -83,10 +138,65 @@ fn publish(matches: &ArgMatches) -> Result<()> {
             println!();
         }
 
-        save_parameter_map(&parameter_map, &json)?;
+        write_parameter_map_to_disk(&parameter_map, &json)?;
     } else {
-        println!("no parameters to publish");
+        println!("no files to publish");
     }
 
     Ok(())
 }
+
+fn get_filenames_in_cache_dir() -> Result<Vec<String>> {
+    let path = parameter_cache_dir();
+
+    if path.exists() {
+        Ok(read_dir(path)?
+            .map(|f| f.unwrap().path())
+            .filter(|p| p.is_file())
+            .map(|p| {
```

Comment:
[`filter_map`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.filter_map) is your friend

```rust
read_dir(path)?.filter_map(|f| {
	let f = f?;
	if !f.path().is_file() {
		return None;
	}
	let file_name = f.path().file_name().unwrap().to_str().unwrap().to_string();
	Some(Ok(file_name))
}).collect()
```

Reply:
This is more code from the original implementation. I'm inclined to leave it as-is for now.